### PR TITLE
Fix issues with route prefix stripping 

### PR DESCRIPTION
**Pull request by ganemone**
#244

### CODE REVIEW

File: test/cli/build.js [hunk 1]
```diff
@@ -138,6 +138,28 @@ test('`fusion build` works in production with default asset path and supplied RO
...
+  const {proc, port} = await start(`--dir=${dir}`, {
+    env: Object.assign({}, process.env, {ROUTE_PREFIX: '/test-prefix'}),
+  });
+  const rootRes = await request(`http://localhost:${port}/`);
+  t.equal(
+    rootRes,
+    'ROOT REQUEST',
+    'strips route prefix correctly for root requests'
+  );
+  const testRes = await request(`http://localhost:${port}/test`);
```
Comment (AlexMSmithCA) [inline test/cli/build.js]: Shouldn't these have the route prefix in the request to ensure it gets stripped?

Comment (ganemone) [reply test/cli/build.js]: yikes - I'm a fool

File: test/fixtures/prefix/src/main.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import App from 'fusion-core';
+
+export default async function() {
+  const app = new App('element', el => el);
+  app.middleware((ctx, next) => {
+    if (ctx.url === '/' && ctx.path === '/') {
+      ctx.body = 'ROOT REQUEST';
+    } else if (ctx.url === '/test' && ctx.path === '/test') {
```
Comment (AlexMSmithCA) [inline test/fixtures/prefix/src/main.js]: We aren't stripping any route prefixes from `path`, right?

Comment (ganemone) [reply test/fixtures/prefix/src/main.js]: It should strip it from any url related props on ctx, since I strip it before sending req/res to koa

**Pull request merged**